sql/src/main/java/io/crate/expression/symbol/Function.java

```diff
@@ -23,37 +23,50 @@
 package io.crate.expression.symbol;
 
 import com.google.common.base.Preconditions;
-import com.google.common.collect.ImmutableList;
 import io.crate.expression.scalar.arithmetic.ArrayFunction;
 import io.crate.metadata.FunctionIdent;
 import io.crate.metadata.FunctionInfo;
 import io.crate.planner.ExplainLeaf;
 import io.crate.types.ArrayType;
 import io.crate.types.DataType;
+import org.elasticsearch.Version;
+import org.elasticsearch.common.Nullable;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
+import java.util.Objects;
 
 public class Function extends Symbol implements Cloneable {
 
     private final List<Symbol> arguments;
     private final FunctionInfo info;
+    private final Symbol filter;
```

Comment:
Maybe also annotate with Nullable.

Although I'm generally wondering if we should go with `Literal.BOOLEAN_TRUE`  as a default value (and do identity checks to figure out if it is a no-op/it is unset).

May be a bit hacky, so I'm not sure - but might also allow us to avoid some branching and have single code paths elsewhere.

Reply:
`Literal.BOOLEAN_TRUE` was my initial approach, but then I went with `null`, coz that might be a bit cheaper serialization-wise.

To check whether it is no-op/unset we would also need to have some serializable property in `Function` or smth like that?

Reply:
ah right, the identity check wouldn't work post streaming. We'd have to go with regular literal/evaluates-to-true.

Let's keep it at `null` then, since it is also less ambigious and we can properly distinguish between "not set" and "true"